src/client/components/dimension-measure-panel/dimension-measure-panel.tsx

```diff
@@ -39,45 +40,93 @@ export interface DimensionMeasurePanelProps {
 }
 
 export interface DimensionMeasurePanelState {
+  dividerPosition: number;
+  containerHeight: number;
+}
+
+function dividerConstraints(height: number) {
+  const minDividerPosition = Math.min(MIN_PANEL_SIZE, height);
+  const maxDividerPosition = Math.max(height - MIN_PANEL_SIZE, 0);
+  return { minDividerPosition, maxDividerPosition };
+}
+
+export function initialPosition(height: number, dataCube: DataCube) {
+  const dimensionsCount = dataCube.dimensions.size();
+  const measuresCount = dataCube.measures.size();
+  const ratio = dimensionsCount / (measuresCount + dimensionsCount);
+
+  const { minDividerPosition, maxDividerPosition } = dividerConstraints(height);
+  return clamp(height * ratio, minDividerPosition, maxDividerPosition);
 }
 
 export class DimensionMeasurePanel extends React.Component<DimensionMeasurePanelProps, DimensionMeasurePanelState> {
 
+  state: DimensionMeasurePanelState = {
+    containerHeight: 2 * MIN_PANEL_SIZE,
+    dividerPosition: MIN_PANEL_SIZE
+  };
+
+  containerRef: Element = null;
+
+  getInitialState = (container: Element) => {
+    if (!container) return;
+
+    this.containerRef = container;
+    const { height: containerHeight } = this.containerRef.getBoundingClientRect();
+    const dividerPosition = initialPosition(containerHeight, this.props.essence.dataCube);
+
+    this.setState({ dividerPosition, containerHeight });
```

Comment:
Setting state in ref callback is considered anti pattern, this logic should be probably moved to `componentDidMount`

Reply:
Why? https://github.com/facebook/react/issues/6557#issuecomment-213022210

Reply:
Last comment in this discussion mentions that it is 

> supported but discouraged

Reply:
I'm still not convinced why. In this three it's stated that reading children dimensions is valid usage of ref callback and setState. What are the downsides?